cpu/atmega_common/include/arduino_arch.hpp

```diff
@@ -0,0 +1,307 @@
+        return &PCMSK0;
+    }
+    return NULL;
+}
+
+/**
+ * @brief   Return the bit in the PCMSK register for the given digital pin
+ */
+static inline uint8_t digitalPinToPCMSKbit(int pin)
+{
+    if ((pin >= 8) && (pin <= 11)) {
+        return pin - 4;
+    }
+    else if (pin == 14) {
+        return 3;
+    }
+    else if (pin == 15) {
+        return 1;
+    }
+    else if (pin == 16) {
+        return 2;
+    }
+    else if (pin == 17) {
+        return 0;
+    }
+    else {
+        return pin - ARDUINO_PIN_A8 + 4;
+    }
+}
+
+/**
+ * @brief   Return the interrupt number for the given digital pin
+ */
+static inline int8_t digitalPinToInterrupt(int pin)
+{
+    if (pin == 0) {
+        return 2;
+    }
+    else if (pin == 1) {
+        return 3;
+    }
+    else if (pin == 2) {
+        return 1;
+    }
+    else if (pin == 3) {
+        return 0;
+    }
+    else if (pin == 7) {
+        return 4;
+    }
+    else {
+        return -1;
+    }
+}
+
+#elif defined(BOARD_ARDUINO_MEGA2560)
+
+/**
+ * @brief   Return the PCICR register for the given digital pin
+ */
+static inline volatile uint8_t *digitalPinToPCICR(int pin)
+{
+    if (((pin >= 10) && (pin <= 13))
+        || ((pin >= 50) && (pin <= 53))
+        || ((pin >= 62) && (pin <= 69))) {
+        return &PCICR;
+    }
+    return NULL;
+}
+
+/**
+ * @brief   Return the bit in the PCICR register for the given digital pin
+ */
+static inline uint8_t digitalPinToPCICRbit(int pin)
+{
+    if (((pin >= 10) && (pin <= 13)) || ((pin >= 50) && (pin <= 53))) {
+        return 0;
+    }
+    else if ((pin >= 62) && (pin <= 69)) {
+        return 2;
+    }
+    else {
+        return 0;
+    }
+}
+
+/**
+ * @brief   Return the PCMSK register for the given digital pin
+ */
+static inline volatile uint8_t *digitalPinToPCMSK(int pin)
+{
+    if (((pin >= 10) && (pin <= 13)) || ((pin >= 50) && (pin <= 53))) {
+        return &PCMSK0;
+    }
+    else if ((pin >= 62) && (pin <= 69)) {
+        return &PCMSK2;
+    }
+    return NULL;
+}
+
+/**
+ * @brief   Return the bit in the PCMSK register for the given digital pin
+ */
+static inline uint8_t digitalPinToPCMSKbit(int pin)
+{
+    if ((pin >= 10) && (pin <= 13)) {
+        return pin - 6;
+    }
+    else if (pin == 50) {
+        return 3;
+    }
+    else if (pin == 51) {
+        return 2;
+    }
+    else if (pin == 52) {
+        return 1;
+    }
+    else if (pin == 53) {
+        return 0;
+    }
+    else if ((pin >= 62) && (pin <= 69)) {
+        return pin - 62;
+    }
+    else {
+        return 0;
+    }
+}
+
+/**
+ * @brief   Return the interrupt number for the given digital pin
+ */
+static inline int8_t digitalPinToInterrupt(int pin)
+{
+    if (pin == 2) {
+        return 0;
+    }
+    else if (pin == 3) {
+        return 1;
+    }
+    else if ((pin >= 18) && (pin <= 21)) {
+        return 23 - pin;
+    }
+    else {
+        return -1;
+    }
+}
+
+#elif defined(BOARD_WASPMOTE_PRO)
```

Comment:
Why is it empty for waspmote-pro ?

Reply:
Because otherwise unrelated programs fail, for example:

```
$ BOARD=waspmote-pro make -C tests/sys_arduino
[...]
/home/jdavid/sandboxes/UiO/RIOT/cpu/atmega_common/include/arduino_arch.hpp: In function ‘volatile uint8_t* digitalPinToPCMSK(int)’:
/home/jdavid/sandboxes/UiO/RIOT/cpu/atmega_common/include/arduino_arch.hpp:257:17: error: ‘PCMSK2’ was not declared in this scope; did you mean ‘PCMSK1’?
  257 |         return &PCMSK2;
      |                 ^~~~~~
      |                 PCMSK1
```

In Arduino `digitalPinToXXX` are macros, when I changed them to inline functions (as requested by @maribu) these errors started.

So in this PR the `arduino_sdi_12` package is not supported for the `waspmote-pro` board. Should I add a comment?

Reply:
Wait, I just realized that this is bound to break things when module `atmega_pcint` is used, as RIOT already registers ISRs for pin changed interrupts. It makes more sense to define `SDI12_EXTERNAL_PCINT` in the pkg and drop these helpers all together.